sql/src/main/java/io/crate/operation/collect/collectors/LuceneBatchIterator.java

```diff
@@ -220,9 +222,17 @@ private void onDoc(int doc, LeafReader reader) throws IOException {
         }
     }
 
-    private void checkClosed() {
+    private void raiseIfClosedOrKilled() {
         if (closed) {
             throw new IllegalStateException("BatchIterator is closed");
         }
+        if (killed != null) {
+            Exceptions.rethrowUnchecked(killed);
+        }
+    }
+
+    @Override
+    public void kill(@Nullable Throwable throwable) {
```

Comment:
Why is the throwable nullable for iterators? If there's no exceptional reason to kill an iterator close should be used right?